Bottom: 223dbf3dcb3c2732004866265a44386144a72124
Top:    54334f4ac4cb838e32f155f5e95edd1352f63684
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-07-04 03:03:31 +0100

parse_y.y: Try to avoid two shift/reduce and reduce/reduce warnings


---

diff --git a/src/parse_y.y b/src/parse_y.y
index 46058e3..b56bcbc 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -213,7 +213,6 @@ parsepcb
 		| {
 		    if (yyPCB != NULL)
 		      {
-			/* This case is when we load a footprint with file->open, or from the command line */
 			yyFont = &yyPCB->Font;
 			yyData = yyPCB->Data;
 			yyData->pcb = yyPCB;
@@ -226,10 +225,9 @@ parsepcb
 		    ElementType *e;
 		    if (yyPCB != NULL)
 		      {
-			/* This case is when we load a footprint with file->open, or from the command line */
 			CreateNewPCBPost (yyPCB, 0);
 			ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, &yyData->LayerN);
-			e = yyPCB->Data->Element->data; /* we know there's only one */
+			e = yyPCB->Data->Element->data;
 			PCB = yyPCB;
 			MoveElementLowLevel (yyPCB->Data, e, -e->BoundingBox.X1, -e->BoundingBox.Y1);
 			PCB = pcb_save;
@@ -706,7 +704,6 @@ pcbdefinition
 		| rats
 		| layer
 		| element
-		| error { YYABORT; }
 		;
 
 via
@@ -1265,91 +1262,6 @@ definitions which use absolute coordinates.
 %end-doc */
 
 element
-		: element_oldformat
-		| element_1.3.4_format
-		| element_newformat
-		| element_1.7_format
-		| element_hi_format
-		;
-
-element_oldformat
-			/* element_flags, description, pcb-name,
-			 * text_x, text_y, text_direction, text_scale, text_flags
-			 */
-		: T_ELEMENT '(' STRING STRING measure measure INTEGER ')' '('
-			{
-				yyElement = CreateNewElement(yyData, yyFont, NoFlags(),
-					$3, $4, NULL, OU ($5), OU ($6), $7, 100, NoFlags(), false);
-				free ($3);
-				free ($4);
-				pin_num = 1;
-			}
-		  elementdefinitions ')'
-			{
-				SetElementBoundingBox(yyData, yyElement, yyFont);
-			}
-		;
-
-element_1.3.4_format
-			/* element_flags, description, pcb-name,
-			 * text_x, text_y, text_direction, text_scale, text_flags
-			 */
-		: T_ELEMENT '(' INTEGER STRING STRING measure measure measure measure INTEGER ')' '('
-			{
-				yyElement = CreateNewElement(yyData, yyFont, OldFlags($3),
-					$4, $5, NULL, OU ($6), OU ($7), IV ($8), IV ($9), OldFlags($10), false);
-				free ($4);
-				free ($5);
-				pin_num = 1;
-			}
-		  elementdefinitions ')'
-			{
-				SetElementBoundingBox(yyData, yyElement, yyFont);
-			}
-		;
-
-element_newformat
-			/* element_flags, description, pcb-name, value, 
-			 * text_x, text_y, text_direction, text_scale, text_flags
-			 */
-		: T_ELEMENT '(' INTEGER STRING STRING STRING measure measure measure measure INTEGER ')' '('
-			{
-				yyElement = CreateNewElement(yyData, yyFont, OldFlags($3),
-					$4, $5, $6, OU ($7), OU ($8), IV ($9), IV ($10), OldFlags($11), false);
-				free ($4);
-				free ($5);
-				free ($6);
-				pin_num = 1;
-			}
-		  elementdefinitions ')'
-			{
-				SetElementBoundingBox(yyData, yyElement, yyFont);
-			}
-		;
-
-element_1.7_format
-			/* element_flags, description, pcb-name, value, mark_x, mark_y,
-			 * text_x, text_y, text_direction, text_scale, text_flags
-			 */
-		: T_ELEMENT '(' INTEGER STRING STRING STRING measure measure
-			measure measure number number INTEGER ')' '('
-			{
-				yyElement = CreateNewElement(yyData, yyFont, OldFlags($3),
-					$4, $5, $6, OU ($7) + OU ($9), OU ($8) + OU ($10),
-					$11, $12, OldFlags($13), false);
-				yyElement->MarkX = OU ($7);
-				yyElement->MarkY = OU ($8);
-				free ($4);
-				free ($5);
-				free ($6);
-			}
-		  relementdefs ')'
-			{
-				SetElementBoundingBox(yyData, yyElement, yyFont);
-			}
-		;
-
-element_hi_format
 			/* element_flags, description, pcb-name, value, mark_x, mark_y,
 			 * text_x, text_y, text_direction, text_scale, text_flags
 			 */
@@ -1433,50 +1345,6 @@ the mark as part of the Element line.
 
 %end-doc */
 
-elementdefinitions
-		: elementdefinition
-		| elementdefinitions elementdefinition
-		;
-
-elementdefinition
-		: pin_1.6.3_format
-		| pin_newformat
-		| pin_oldformat
-		| pad_newformat
-		| pad
-			/* x1, y1, x2, y2, thickness */
-		| T_ELEMENTLINE '[' measure measure measure measure measure ']'
-			{
-				CreateNewLineInElement(yyElement, NU ($3), NU ($4), NU ($5), NU ($6), NU ($7));
-			}
-			/* x1, y1, x2, y2, thickness */
-		| T_ELEMENTLINE '(' measure measure measure measure measure ')'
-			{
-				CreateNewLineInElement(yyElement, OU ($3), OU ($4), OU ($5), OU ($6), OU ($7));
-			}
-			/* x, y, width, height, startangle, anglediff, thickness */
-		| T_ELEMENTARC '[' measure measure measure measure number number measure ']'
-			{
-				CreateNewArcInElement(yyElement, NU ($3), NU ($4), NU ($5), NU ($6), $7, $8, NU ($9));
-			}
-			/* x, y, width, height, startangle, anglediff, thickness */
-		| T_ELEMENTARC '(' measure measure measure measure number number measure ')'
-			{
-				CreateNewArcInElement(yyElement, OU ($3), OU ($4), OU ($5), OU ($6), $7, $8, OU ($9));
-			}
-			/* x, y position */
-		| T_MARK '[' measure measure ']'
-			{
-				yyElement->MarkX = NU ($3);
-				yyElement->MarkY = NU ($4);
-			}
-		| T_MARK '(' measure measure ')'
-			{
-				yyElement->MarkX = OU ($3);
-				yyElement->MarkY = OU ($4);
-			}
-		| { attr_list = & yyElement->Attributes; } attribute
-		;
 
 relementdefs
 		: relementdef
@@ -1575,51 +1443,7 @@ pin_1.7_format
 			}
 		;
 
-pin_1.6.3_format
-			/* x, y, thickness, drilling hole, name, number, flags */
-		: T_PIN '(' measure measure measure measure STRING STRING INTEGER ')'
-			{
-				CreateNewPin(yyElement, OU ($3), OU ($4), OU ($5), 2*GROUNDPLANEFRAME,
-					OU ($5) + 2*MASKFRAME, OU ($6), $7, $8, OldFlags($9));
-				free ($7);
-				free ($8);
-			}
-		;
 
-pin_newformat
-			/* x, y, thickness, drilling hole, name, flags */
-		: T_PIN '(' measure measure measure measure STRING INTEGER ')'
-			{
-				char	p_number[8];
-
-				sprintf(p_number, "%d", pin_num++);
-				CreateNewPin(yyElement, OU ($3), OU ($4), OU ($5), 2*GROUNDPLANEFRAME,
-					OU ($5) + 2*MASKFRAME, OU ($6), $7, p_number, OldFlags($8));
-
-				free ($7);
-			}
-		;
-
-pin_oldformat
-			/* old format: x, y, thickness, name, flags
-			 * drilling hole is 40% of the diameter
-			 */
-		: T_PIN '(' measure measure measure STRING INTEGER ')'
-			{
-				Coord	hole = OU ($5) * DEFAULT_DRILLINGHOLE;
-				char	p_number[8];
-
-					/* make sure that there's enough copper left */
-				if (OU ($5) - hole < MIN_PINORVIACOPPER && 
-					OU ($5) > MIN_PINORVIACOPPER)
-					hole = OU ($5) - MIN_PINORVIACOPPER;
-
-				sprintf(p_number, "%d", pin_num++);
-				CreateNewPin(yyElement, OU ($3), OU ($4), OU ($5), 2*GROUNDPLANEFRAME,
-					OU ($5) + 2*MASKFRAME, hole, $6, p_number, OldFlags($7));
-				free ($6);
-			}
-		;
 
 /* %start-doc pcbfile Pad
 
@@ -1683,30 +1507,6 @@ pad_1.7_format
 			}
 		;
 
-pad_newformat
-			/* x1, y1, x2, y2, thickness, name , pad number, flags */
-		: T_PAD '(' measure measure measure measure measure STRING STRING INTEGER ')'
-			{
-				CreateNewPad(yyElement,OU ($3),OU ($4),OU ($5),OU ($6),OU ($7), 2*GROUNDPLANEFRAME,
-					OU ($7) + 2*MASKFRAME, $8, $9, OldFlags($10));
-				free ($8);
-				free ($9);
-			}
-		;
-
-pad
-			/* x1, y1, x2, y2, thickness, name and flags */
-		: T_PAD '(' measure measure measure measure measure STRING INTEGER ')'
-			{
-				char		p_number[8];
-
-				sprintf(p_number, "%d", pin_num++);
-				CreateNewPad(yyElement,OU ($3),OU ($4),OU ($5),OU ($6),OU ($7), 2*GROUNDPLANEFRAME,
-					OU ($7) + 2*MASKFRAME, $8,p_number, OldFlags($9));
-				free ($8);
-			}
-		;
-
 flags		: INTEGER	{ $$ = OldFlags($1); }
 		| STRING	{ $$ = string_to_flags ($1, yyerror); }
 		;
